automation_script/ut_results_parse_automation.py

```diff
@@ -0,0 +1,402 @@
+                res_default_priority = run_priority_tests("default", overall_logs_path_current_run, test_reports_src)
+                res_all_tests_dict["default"] = res_default_priority
+            if "distributed" in workflow_list:
+                res_distributed_priority = run_priority_tests("distributed", overall_logs_path_current_run, test_reports_src)
+                res_all_tests_dict["distributed"] = res_distributed_priority
+            if "inductor" in workflow_list:
+                print("Inductor priority tests cannot run since no core tests defined with inductor workflow.")
+    # Run specified tests for each workflow
+    elif (args.default_list or args.distributed_list or args.inductor_list) and not args.test_config and not args.priority_test:
+        if args.default_list:
+            default_workflow_list = []
+            for item in args.default_list:
+                default_workflow_list.append(item)
+            res_default_selected = run_selected_tests("default", overall_logs_path_current_run, test_reports_src, default_workflow_list)
+            res_all_tests_dict["default"] = res_default_selected
+        if args.distributed_list:
+            distributed_workflow_list = []
+            for item in args.distributed_list:
+                distributed_workflow_list.append(item)
+            res_distributed_selected = run_selected_tests("distributed", overall_logs_path_current_run, test_reports_src, distributed_workflow_list)
+            res_all_tests_dict["distributed"] = res_distributed_selected
+        if args.inductor_list:
+            inductor_workflow_list = []
+            for item in args.inductor_list:
+                 inductor_workflow_list.append(item)
+            res_inductor_selected = run_selected_tests("inductor", overall_logs_path_current_run, test_reports_src, inductor_workflow_list)
+            res_all_tests_dict["inductor"] = res_inductor_selected
+    else:
+        raise Exception("Invalid test configurations!")
+
+    # restore environment variables
+    os.environ.clear()
+    os.environ.update(_environ)
+
+    return res_all_tests_dict
+
+def parse_args():
+    parser = argparse.ArgumentParser(description='Run PyTorch unit tests and generate xml results summary')
+    parser.add_argument('--test_config', nargs='+', default=[], type=str, help="test workflows to be executed")
+    parser.add_argument('--priority_test', action='store_true', help="run priority tests only")
+    parser.add_argument('--default_list', nargs='+', default=[], help="default tests to be executed")
+    parser.add_argument('--distributed_list', nargs='+', default=[], help="distributed tests to be executed")
+    parser.add_argument('--inductor_list', nargs='+', default=[], help="inductor tests to be executed")
+    return parser.parse_args()
+
+def main():
+    global args
+    args = parse_args()
+    all_tests_results = {}
+    all_tests_results = run_test_and_summarize_results(args)
```

Comment:
@BLOrange-AMD These scripts need to support two use-cases:
1. QA Automation use-case: This would involve invoking the logic from a python script and post-processing the returned results. For this, one would need to be able to import the `run_test_and_summarize_results` function and pass it an `args` object which would be the equivalent of the output of `parse_args()` above.
2. General unit test run use-case: This would involve invoking these scripts from the command line. For this, one can invoke this script with command-line args.

Let's make sure we support both use-cases.